src/bindings/python/wheel/CMakeLists.txt

```diff
@@ -100,31 +100,14 @@ set(openvino_wheel_path "${openvino_wheels_output_dir}/${openvino_wheel_name}")
 # create target for openvino.wheel
 #
 
-execute_process(COMMAND ${Python3_EXECUTABLE} -m pip --version
-                OUTPUT_VARIABLE pip_version OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-string(REGEX MATCH "pip[ ]+([\\.0-9]*)" pip_version "${pip_version}")
-set(pip_version ${CMAKE_MATCH_1})
-
-if(pip_version VERSION_GREATER_EQUAL 22.0)
-    set(wheel_build_command
-        ${Python3_EXECUTABLE} -m pip wheel
-            --no-deps
-            --wheel-dir ${openvino_wheels_output_dir}
-            --verbose
-            --build-option --build-number=${WHEEL_BUILD}
-            --build-option --plat-name=${PLATFORM_TAG}
-            "${CMAKE_CURRENT_SOURCE_DIR}")
-else()
-    set(wheel_build_command
-        ${Python3_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/setup.py"
-            --quiet
-            --no-user-cfg
-            bdist_wheel
-            --dist-dir ${openvino_wheels_output_dir}
-            --build-number=${WHEEL_BUILD}
-            --plat-name=${PLATFORM_TAG})
-endif()
+# for --config-setting explanation see https://github.com/pypa/setuptools/issues/2491
+set(wheel_build_command
+    ${Python3_EXECUTABLE} -m build "${CMAKE_CURRENT_SOURCE_DIR}"
```

Comment:
BTW, should we use `pip wheel` instead ? to avoid dependency on `build`

Reply:
BTW, are you going to rework this call as well ?
https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/CMakeLists.txt#L328

Reply:
> BTW, should we use pip wheel instead ? to avoid dependency on build

I tried making `pip wheel` work for both ranges of `pip` versions (in the previous code there was an `if` depending on `pip` version), but the old ones

In the original code there was an `if` depending on `pip` version, and for older `pip`s it would call `setup.py` directly. I tried unifying the functionality to call `pip wheel` for all `pip` versions, but it seems there's some incompatibility which results in very unfriendly errors which would (in my opinion) a lot of changes, only once I figure out what needs to be changed.

Changing to using `build` (which is a [pypa recommended replacement](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#what-commands-should-be-used-instead)) works for all pip versions and removes the branching based on `pip` version.

We could work on making `pip wheel` work with old pip versions, but I believe it will be time consuming and difficult, or stay with `pip wheel` but drop support for older `pip`s.

> BTW, are you going to rework this call as well ?
https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/CMakeLists.txt#L328

The changes regarding this call are in https://github.com/openvinotoolkit/openvino/pull/27227.

Reply:
we can enable pip wheel for newer pip versions, while old ones (via `build`) will not be used after some time and we can just force to use newer pip only (once Ubuntu 20.04 is dropped, for example)